syntax = "proto2";

package org.copdai;

message Data {
    repeated Geolocation geolocations = 1;
    repeated Accelerometer accelerometers = 2;
    repeated Obstacles obstacles = 3;
    required double time = 25;

}

message Geolocation {
    required double lat = 1;
    required double lng = 2;
    optional double z = 3;
    required double time = 25;
}

message Accelerometer {
      required double rotation_x = 1;
      required double rotation_y = 2;
      required double rotation_z = 3;
      required double velocity_x = 8;
      required double velocity_y = 9;
      required double velocity_z = 10;
      required double time = 25;
}

message Obstacles {
    repeated ObstacleCoordinate obstacles_coordinates = 1;
    required double time = 25;
}

message ObstacleCoordinate {
    required double x1 = 1;
    required double y1 = 2;
    required double z1 = 3;

    required double x2 = 8;
    required double y2 = 9;
    required double z2 = 10;
}

message Action {
    required double acceleration = 1;
    required double torque = 2;
    required double time = 25;
}